File: code_corpus/manual/sort_5b.java
public int[] merge(int[] L, int[] R){
  int lenL = L.length;
  int lenR = R.length;
  int[] merged = new int[lenL+lenR];
  int i = 0;
  int j = 0;
  while(i<lenL||j<lenR){
    if(i<lenL & j<lenR){
      if(L[i]<=R[j]){
        merged[i+j] = L[i];
        i++;
      }
      else{
        merged[i+j] = R[j];
        j++;
      }
    }
    else if(i<lenL){
      merged[i+j] = L[i];
      i++;
    }
    else if(j<lenR){
      merged[i+j] = R[j];
      j++;
     }
   }
   return merged;
}

File: code_corpus/manual/sort_5b.m
function dataOut = merge(L,R)
lenL = size(L,2);
lenR = size(R,2);
i = 0;
j = 0;
merged = [];
while(i<lenL||j<lenR)
  if (i<lenL && j<lenR)
    if(L(i+1)<=R(j+1))
      merged(i+j+1) = L(i+1);
      i = i+1;
    else
      merged(i+j+1) = R(j+1);
      j = j+1;
    end
  elseif(i<lenL)
    merged(i+j+1) = L(i+1);
    i = i+1;
  elseif(j<lenR)
    merged(i+j+1) = R(j+1);
    j = j+1;
  end
end
dataOut = merged;
